components/invitation_modal/invitation_modal_members_step.jsx

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {isEmail} from 'mattermost-redux/utils/helpers';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import InviteIcon from 'components/svg/invite_icon';
+import UsersEmailsInput from 'components/widgets/inputs/users_emails_input.jsx';
+
+import BackIcon from 'components/svg/back_icon';
+
+import {getSiteURL} from 'utils/url.jsx';
+import debouncePromise from 'utils/debounce_promise.jsx';
+import {t} from 'utils/i18n.jsx';
+
+import './invitation_modal_members_step.scss';
+
+export default class InvitationModalMembersStep extends React.Component {
+    static propTypes = {
+        inviteId: PropTypes.string.isRequired,
+        goBack: PropTypes.func,
+        searchProfiles: PropTypes.func.isRequired,
+        onEdit: PropTypes.func.isRequired,
+        onSubmit: PropTypes.func.isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            usersAndEmails: [],
+        };
+    }
+
+    usersLoader = async (term) => {
+        if (isEmail(term)) {
+            return [];
+        }
+        const {data} = await this.props.searchProfiles(term);
+        return data;
+    }
+    debouncedUsersLoader = debouncePromise(this.usersLoader, 150);
+
+    onChange = (usersAndEmails) => {
+        this.setState({usersAndEmails});
+        this.props.onEdit();
+    }
+
+    submit = () => {
+        const users = [];
+        const emails = [];
+        for (const userOrEmail of this.state.usersAndEmails) {
+            if (isEmail(userOrEmail)) {
+                emails.push(userOrEmail);
+            } else {
+                users.push(userOrEmail);
+            }
+        }
+        this.props.onSubmit(users, emails);
+    }
+
+    render() {
+        const inviteUrl = getSiteURL() + '/signup_user_complete/?id=' + this.props.inviteId;
```

Comment:
Not required but `getSiteURL` and could probably be done just once in the constructor.

Reply:
Yes, I think I prefer to maintain it here because is not an expensive operation and I think there is more readable.